Lib/fontTools/cffLib/specializer.py

```diff
@@ -417,12 +559,24 @@ def specializeCommands(commands,
 				continue
 
 			# Merge adjacent hlineto's and vlineto's.
+			# In CFF2 charstrings from variable fonts, each
+			# arg item may be a list of blendable values, one from
+			# each source font.
 			if (i and op in {'hlineto', 'vlineto'} and
-					(op == commands[i-1][0]) and
-					(not isinstance(args[0], list))):
+							(op == commands[i-1][0])):
 				_, other_args = commands[i-1]
 				assert len(args) == 1 and len(other_args) == 1
-				commands[i-1] = (op, [other_args[0]+args[0]])
+				arg0 = args[0]
+				arg1 = other_args[0]
+				if isinstance(arg0, list) and isinstance(arg1, list):
+					new_args = [[a1 + a2 for a1, a2 in zip(arg0, arg1)]]
+				elif isinstance(arg0, list):
+					new_args = _combineLineArgs(arg0, arg1)
+				elif isinstance(arg1, list):
+					new_args = _combineLineArgs(arg1, arg0)
+				else:
+					new_args = [arg0 + arg1]
```

Comment:
This block should simply become `new_args = [_addArgs(args[0], other_args[0])]

Reply:
see below